src/chicago_participatory_urbanism/geocoder_local.py

```diff
@@ -0,0 +1,139 @@
+import pandas as pd
+import importlib.metadata
+import logging
+import geopandas as gpd
+from shapely.ops import unary_union
+from shapely.geometry import Point, MultiPoint, LineString, MultiLineString
+from src.chicago_participatory_urbanism.location_structures import StreetAddress, Intersection
+
+# address point csv from https://hub-cookcountyil.opendata.arcgis.com/datasets/5ec856ded93e4f85b3f6e1bc027a2472_0/about
+try:
```

Comment:
I would highly recommend putting this code block in a global scope like this for a few reasons:

* It will initialize variables and objects in the global scope
* it will re-initialize these variables every time the Python module is imported, which could be multiple times throughout the lifecycle of a script
* it initializes them before the `Geocoder` object is actually created, even if the Geocoder class isn't used by the code. This reduces performance and can also make it much more difficult to write tests. 

Instead, consider putting this within an `__init__` method of the `Geocoder` class itself, and use `self` to save variables like `df`. This will make sure that it only happens once - when the Geocoder is initialized. 

example: https://github.com/suryadutta/chihacknight-ward-wise-data-analysis/blob/2ca215a247e841e559594c8c3cd2568cd4de1d88/src/chicago_participatory_urbanism/geocoder_api.py#L19-L23